kernel/aster-nix/src/time/clocks/system_wide.rs

```diff
@@ -0,0 +1,300 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use alloc::sync::Arc;
+use core::time::Duration;
+
+use aster_frame::{
+    arch::timer::{self, Jiffies},
+    cpu_local,
+    sync::SpinLock,
+    CpuLocal,
+};
+use aster_time::read_monotonic_time;
+use paste::paste;
+use spin::Once;
+
+use crate::time::{system_time::START_TIME_AS_DURATION, timer::TimerManager, Clock, SystemTime};
+
+struct PrivateFieldMarker;
+
+/// The Clock that reads the jiffies, and turn the counter into `Duration`.
+pub struct JiffiesClock {
+    _marker: PrivateFieldMarker,
```

Comment:
I think defining `PrivateFieldMarker` is an overkill. The following form is pretty clear, rigtht?

```rust
pub struct JiffiesClock {
    _private: (),
}
```